src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionSettings.cs

```diff
@@ -72,6 +73,7 @@ public HttpConnectionSettings CloneAndNormalize()
                 _connectTimeout = _connectTimeout,
                 _credentials = _credentials,
                 _defaultProxyCredentials = _defaultProxyCredentials,
+                _defaultCredentials = _defaultCredentials,
```

Comment:
This field should not be added. It is confusing. And it isn't being used as originally intended.

The `_defaultProxyCredentials` has nothing to do with CredentialCache.DefaultCredentials. In hindsight, I wish I had named the HttpClientHandler.DefaultProxyCredentials as HttpClientHandler.SystemConfiguredProxyCredentials.

The `defaultProxyCredentials` on the handler represents the ICredentials that should be associated with a system-default proxy. A system-default proxy is configured/discovered with IESettings (Windows), Environment Variables (Windows/Linux/Mac) or MacProxySettings (Mac).

But adding a field called `_defaultCredentials` which is essentially a shortcut to CredentialCache.DefaultCredentials doesn't save  any performance. And it is really confusing since `_defaultProxyCredentials` means something so different.

So, my suggestion is to get rid of this new field, `_defaultCredentials`

Reply:
You are right I misread the original comment. The whole defaultCredentialUsed expression must be precalculated.